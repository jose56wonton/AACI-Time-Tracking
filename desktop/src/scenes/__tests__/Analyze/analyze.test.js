import React from 'react';
import { shallow } from 'enzyme';

import AnalyzeHOC,{ Analyze } from 'scenes/Analyze/analyze';

const props =  {
  classes: {},
  history: {
    goBack: jest.fn()
  }
};

const setup = overRides => {  
  return shallow(<Analyze {...props} {...overRides}/>);
};

const setupHOC = overRides => {
  return shallow(<AnalyzeHOC {...props} {...overRides} />);
};

describe('Analyze Scene', () => {  
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
  it('should display different containers based on state.tabValue', () =>{
    const wrapper = setup();
    const instance = wrapper.instance();
    expect(wrapper.state().tabValue).toEqual(1);
    instance.handleTabValueChange(null,2);
    expect(wrapper.state().tabValue).toEqual(2);
    instance.handleTabValueChange(null,0);
    expect(wrapper.state().tabValue).toEqual(0);
    instance.handleTabValueChange(null,3);
    expect(wrapper.state().tabValue).toEqual(3);
  });
  it('should call this.props.history.goBack() on this.back()', () => {
    const wrapper = setup();
    const instance = wrapper.instance();
    expect(props.history.goBack).toHaveBeenCalledTimes(0);
    instance.back();
    expect(props.history.goBack).toHaveBeenCalledTimes(1);
  });
});