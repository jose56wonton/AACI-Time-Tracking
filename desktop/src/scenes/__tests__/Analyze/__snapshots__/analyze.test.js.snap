// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Signin Scene should render correctly 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Analyze Scene should render correctly 1`] = `
<div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Tabs)
        onChange={[Function]}
        value={1}
      >
        <WithStyles(Tab)
          label="Employees"
        />
        <WithStyles(Tab)
          label="Projects"
        />
        <WithStyles(Tab)
          label="Tasks"
        />
        <WithStyles(Tab)
          label="Shifts"
        />
      </WithStyles(Tabs)>
      <WithStyles(IconButton)
        color="inherit"
        onClick={[Function]}
      >
        <pure(ArrowBackIcon) />
      </WithStyles(IconButton)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={4}
    >
      <Connect(ProjectIndexContainer) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={8}
    >
      <Connect(ProjectContainer) />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`Analyze Scene should render correctly withStyles 1`] = `
<Route>
  <Component />
</Route>
`;
